File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort(key=lambda x: (x[1], x[0]))
        res = 1
        last = points[0][1]
        for st,ed in points:
            if st <= last <= ed:
                pass
            else:
                last = ed
                res += 1
        return res

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, Comparator.comparingInt(o -> o[1]));
        int res = 1;
        int last = points[0][1];
        for (int[] point : points) {
            if (point[0] <= last && last <= point[1]) {
                continue;
            }
            last = point[1];
            res++;
        }
        return res;
    }
}